src/coreclr/interop/comwrappers.cpp

```diff
@@ -736,7 +736,14 @@ HRESULT ManagedObjectWrapper::QueryInterface(
                     return E_NOINTERFACE;
 
                 default:
+#if defined(__clang__) && (__clang_major__ < 14)
+                    // NOTE: here the `result == *` of assert is just for avoiding the compiling error
+                    // `fallthrough annotation in unreachable code` within debug mode using clang <=13.
+                    // Because clang versions before 14 had a bug here.
+                    _ASSERTE((result == TryInvokeICustomQueryInterfaceResult::FailedToInvoke) && "Unknown result value");
+#else
                     _ASSERTE(false && "Unknown result value");
+#endif
```

Comment:
```suggestion
_ASSERTE((result == TryInvokeICustomQueryInterfaceResult::FailedToInvoke) && "Unknown result value");
```

Let's just do the thing that should work in all cases rather than the noise.